polling_stations/apps/data_importers/management/commands/import_eoni_from_s3.py

```diff
@@ -0,0 +1,415 @@
+
+            # Validate file
+            self.validate_file()
+
+            # # Set import scheme
+            self.set_import_scheme(options)
+
+            # Set outpath for reprojected file
+            self.set_output_path(options)
+
+            # Create reprojected file at output path
+            self.create_reprojected_file()
+
+            # # Run the import
+            self.call_import_eoni_command(options)
+
+        finally:
+            # Cleanup
+            self.cleanup(options)
+
+    def set_s3_key_and_wrapper(self, options):
+        # Set bucket_name  based on input method
+        if options.get("input_uri"):
+            self.stdout.write("Setting s3 config from input_uri")
+            bucket_name, input_key = parse_s3_uri(options["input_uri"])
+            self.bucket_name = bucket_name
+            self.input_key = input_key
+            self.s3_wrapper = S3Wrapper(bucket_name=self.bucket_name)
+
+            return
+
+        # If no input URI is provided, require DC_ENVIRONMENT for default bucket_name
+        if not self.dc_environment:
+            raise CommandError(
+                "Either --input-uri or DC_ENVIRONMENT environment variable must be set"
+            )
+        self.stdout.write(
+            "Inferring s3 config from DC_ENVIRONMENT environment variable"
+        )
+        self.stdout.write(f"DC_ENVIRONMENT: {self.dc_environment}")
+
+        self.bucket_name = f"eoni-data.wheredoivote.co.uk.{self.dc_environment}"
+        self.s3_wrapper = S3Wrapper(bucket_name=self.bucket_name)
+
+        self.input_key = self.get_latest_file_on_s3()
+
+    def download_file(self):
+        self.stdout.write(
+            f"Fetching latest file:\n\ts3://{self.bucket_name}/{self.input_key}"
+        )
+        self.local_path, downloaded = self.s3_wrapper.download_file(self.input_key)
+        if downloaded:
+            self.stdout.write(f"Downloaded and saved to:\n\t{self.local_path}")
+        else:
+            self.stdout.write(f"Using cached version found at:\n\t{self.local_path}")
+
+    def get_latest_file_on_s3(self):
+        objects = self.s3_wrapper.bucket.objects.all()
+        object_keys = sorted(
+            [obj.key for obj in objects if is_eoni_export_key(obj.key)]
+        )
+
+        if len(object_keys) == 0:
+            raise CommandError(
+                f"Failed to find any EONI export keys for {self.bucket_name}"
+            )
+        return object_keys[-1]
+
+    def decode_csv(self):
+        with open(self.local_path, "rb") as csv_file:
+            body = csv_file.read()
+            try:
+                decoded, csv_encoding = attempt_decode(body)
+                csv_encoding = csv_encoding
+            except UnicodeDecodeError:
+                raise CommandError("Failed to decode CSV using any expected encoding")
+            self.stdout.write(f"{csv_encoding} encoding detected.")
+            self.reader = csv.DictReader(decoded.splitlines())
```

Comment:
I did think about making 
```python
self.rows = list(csv.DictReader(decoded.splitlines())
```

So that we wouldn't get tripped up by iterating the reader and then it being empty. 
But we only need to go through it once at the moment, so have left it like this. 

I guess if we do more validation or reporting on the csv then this could be revisited.

Reply:
I think this is fine as it stands